main.nf

```diff
@@ -115,110 +121,148 @@ process route_file {
         }
     }
     args_str = args_strs.join(' ')
-
     """
-    router.py --type ${type} --file ${file} ${args_str}
+    router.py --type ${type} --file ${file} --stem ${stem} ${args_str}
     """
 }
 
 process Build_config{
-    tag "config"
+    tag "${stem}"
     debug verbose_log
+
     container "hamat/webatlas-build-config:${version}"
     publishDir params.outdir, mode: "copy"
 
     input:
-        val(title)
-        val(dataset)
-        val(url)
-        val(zarr_md)
-        val(files)
-        val(options)
+        tuple val(stem), val(anndata), val(raster), val(label), val(raster_md), val(label_md), val(title), val(dataset), val(url), val(options)
         val(layout)
         val(custom_layout)
 
     output:
-        path("config.json")
+        path("${stem}_config.json")
 
     script:
-    files = files.collect{ /\'/ + it.trim() + /\'/ }
-
-    file_paths = files ? "--file_paths [" + files.join(',') + "]": ""
+    anndata = file(anndata).name
+    raster = file(raster).name
+    lab = file(label).name
     url_str = url?.trim() ? "--url ${url}" : ""
     clayout_str = custom_layout?.trim() ? "--custom_layout \"${custom_layout}\"" : ""
-    zarr_md_str = zarr_md ? "--image_zarr " + /"/ + new JsonBuilder(zarr_md).toString().replace(/"/,/\"/).replace(/'/,/\'/) + /"/ : ""
     options_str = options ? "--options " + /"/ + new JsonBuilder(options).toString().replace(/"/,/\"/).replace(/'/,/\'/) + /"/ : ""
 
     """
     build_config.py \
         --title "${title}" \
         --dataset ${dataset} \
-        ${zarr_md_str} \
-        ${options_str} \
-        ${file_paths} \
+        --image_zarr '{"${raster}":${raster_md}, "${lab}":${label_md}}' \
+        --file_paths '["${anndata}"]' \
         ${url_str} \
+        ${options_str} \
         --layout ${layout} ${clayout_str}
     """
 }
 
-process generate_label_image {
-    tag "${h5ad}"
+process Generate_label_image {
+    tag "${stem}"
     debug verbose_log
+
     container "generate_label:latest"
-    publishDir params.outdir, mode: "copy"
+    storeDir params.outdir
 
     input:
-        path(h5ad)
+        tuple val(stem), val(ome_md_json), path(h5ad), val(data_type), val(args)
 
     output:
-        path("${stem}_with_label.zarr")
+        tuple val(stem), val("label"), file("${stem}.tif")
 
     script:
-    stem = h5ad.baseName
     """
-    generate_label.py --stem "${stem}" --h5ad ${h5ad}
+    generate_label.py --stem "${stem}" --ome_md '${ome_md_json}' --h5ad ${h5ad}
     """
-
 }
 
+
+Channel.fromPath(params.tsv)
+    .splitCsv(header:true, sep:"\t")
+    .multiMap { l ->
+        images: [l.title + "_" + l.dataset, l.image_type, file(l.image_path)]
+        /*data: [l.title + "_" + l.dataset, l.h5ad, file(l.h5ad).extension, l.args]*/
```

Comment:
@dannda also here. I've temporarily commented out the args and forgot to put them back. having plain `l.args` will bug in the string construction. Shall we use `fire` there?